sourcetracker/_plot.py

```diff
@@ -10,16 +10,271 @@
+            ratios.append(1)
+            axes.append("ax" + str(i))
+            g.append("g" + str(i))
+        ratios.append(0.08)
+        axes.append("axcb")
+        fig, axes = plt.subplots(1, len(axes),
+                                 gridspec_kw={'width_ratios': ratios},
+                                 figsize=((prop.shape[1] * 3 / 4)+4,
+                                          (prop.shape[0] * 3 / 4)+4))
+        for i in range(len(prop.columns)):
+            if i == 0:
+                g[i] = sns.heatmap(prop.iloc[:, i:i + 1],
+                                   vmin=0, cmap=self.color,
+                                   cbar=False, annot=annot, ax=axes[i])
+                g[i].set_xlabel("")
+                g[i].set_ylabel(ylabel)
+            elif i == midpoint:
+                g[i] = sns.heatmap(prop.iloc[:, i:i+1], vmin=0,
+                                   cmap=self.color, cbar=False,
+                                   annot=annot, ax=axes[i])
+                g[i].set_xlabel("")
+                g[i].set_ylabel("")
+                g[i].set_yticks([])
+                g[i].set_title(self.title)
+            elif i == len(prop.columns) - 1:
+                g[i] = sns.heatmap(prop.iloc[:, i:i + 1], vmin=0,
+                                   cmap=self.color, annot=annot,
+                                   ax=axes[i],
+                                   cbar_ax=axes[i + 1])
+                g[i].set_xlabel("")
+                g[i].set_ylabel("")
+                g[i].set_yticks([])
+            else:
+                g[i] = sns.heatmap(prop.iloc[:, i:i + 1],
+                                   vmin=0, cmap=self.color,
+                                   cbar=False, annot=annot,
+                                   ax=axes[i])
+                g[i].set_xlabel("")
+                g[i].set_ylabel("")
+                g[i].set_yticks([])
+        for ax in g:
+            tl = ax.get_xticklabels()
+            ax.set_xticklabels(tl, rotation=0)
+            tly = ax.get_yticklabels()
+            ax.set_yticklabels(tly, rotation=0)
+        if normalized:
+            if unknowns:
+                add_line = tra + "_pairedheatmap_normalized.png"
+                plt.savefig(os.path.join(self.file,
+                                         self.title + add_line))
+            else:
+                add_line = tra + "_pairedheatmap_nounknown_normalized.png"
+                plt.savefig(os.path.join(self.file,
+                                         self.title + add_line))
+        else:
+            if unknowns:
+                add_line = tra + "_pairedheatmap.png"
+                plt.savefig(os.path.join(self.file,
+                                         self.title + add_line))
+            else:
+                add_line = tra + "_pairedheatmap_nounknowns.png"
+                plt.savefig(os.path.join(self.file,
+                                         self.title + add_line))
```

Comment:
```suggestion
            if unknowns:
                add_line = tra + "_pairedheatmap_normalized.png"
            else:
                add_line = tra + "_pairedheatmap_nounknown_normalized.png"
        else:
            if unknowns:
                add_line = tra + "_pairedheatmap.png"
            else:
                add_line = tra + "_pairedheatmap_nounknowns.png"
            plt.savefig(os.path.join(self.file, self.title + add_line))
```

Reply:
I think there is a lot of repeat code in this PR. Could you refactor so your branches change the variables and then the function is ran at the end.